sdk/tracing/azure-core-tracing-opencensus/src/main/java/com/azure/core/tracing/opencensus/OpenCensusTracer.java

```diff
@@ -258,7 +256,11 @@ private void addSpanRequestAttributes(Span span, Context context, String spanNam
      * @return The component name contained in the context.
      */
     private static String parseComponentValue(String spanName) {
-        return spanName.substring(spanName.indexOf(".") + 1, spanName.lastIndexOf("."));
+        if (spanName.length() > 0 && spanName.indexOf(".") != -1 && spanName.lastIndexOf(".") != -1) {
```

Comment:
the jit is hopefully going to optimize away these redundant calls to indexOf; but just in case, could you refactor the code to use temporary ints and avoid the duplicate API calls?